### PR TITLE
CI: Run phpunit test one times more on error

### PR DESCRIPTION
**Pull request by nabim777**
## Description

On CI, there was a flaky error in the PHPUnit test related to the mock server port. So In this PR, there is code added for CI to re-run the phpunit test one time whenever there is an error araised. 

**Error:**
```console
There was 1 error:

1) OCA\OpenProject\Service\OpenProjectAPIServiceTest::testGetAvailableOpenProjectProjectsPact
PhpPact\Consumer\Exception\MockServerNotStartedException: The mock server could not be started

/home/runner/work/integration_openproject/integration_openproject/server/apps/integration_openproject/vendor/pact-foundation/pact-php/src/PhpPact/Consumer/Service/MockServer.php:35
/home/runner/work/integration_openproject/integration_openproject/server/apps/integration_openproject/vendor/pact-foundation/pact-php/src/PhpPact/Consumer/Driver/Interaction/InteractionDriver.php:50
/home/runner/work/integration_openproject/integration_openproject/server/apps/integration_openproject/vendor/pact-foundation/pact-php/src/PhpPact/Consumer/InteractionBuilder.php:74
/home/runner/work/integration_openproject/integration_openproject/server/apps/integration_openproject/tests/lib/Service/OpenProjectAPIServiceTest.php:3365

ERRORS!
Tests: 377, Assertions: 677, Errors: 1.

Generating code coverage report in Cobertura XML format ... done [00:00.188]

Generating code coverage report in HTML format ... done [00:00.238]
```

## Related Issue or Workpackage

- https://community.openproject.org/projects/nextcloud-integration/work_packages/57385

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Updated `CHANGELOG.md` file

### CODE REVIEW

File: .github/workflows/shared_workflow.yml [hunk 1]
```diff
@@ -111,10 +111,10 @@ jobs:
           ./occ a:e integration_openproject
           cd apps/integration_openproject
           # The following if block can be removed once Nextcloud no longer supports PHP 8.0
-          if [ "${{matrix.phpVersion}}" -eq 8 ]; then
-            make phpunitforphp8.0
+          if [ "${{ matrix.phpVersion }}" -eq 8 ]; then
+            make phpunitforphp8.0 || (echo "First attempt failed. Retrying PHPUnit tests..." && make phpunitforphp8.0)
```
Comment (nabim777) [inline .github/workflows/shared_workflow.yml:115]:
```suggestion
            make phpunitforphp8.0 || (echo "First attempt failed. Retrying PHPUnit tests..." && make phpunitforphp8.0)
```

If the command before this `||` fails, the command after this pipe `||` will be executed.

Comment (SagarGi) [inline .github/workflows/shared_workflow.yml:115]:
```suggestion
            make phpunitforphp8.0 || (echo "A few of the unit tests were unsuccessful. Rerunning the unit test once again......" && make phpunitforphp8.0)
```

File: .github/workflows/shared_workflow.yml [hunk 2]
```diff
@@ -111,10 +111,10 @@ jobs:
...
           else
-            make phpunit
+            make phpunit || (echo "First attempt failed. Retrying PHPUnit tests..." && make phpunit)
```
Comment (SagarGi) [inline .github/workflows/shared_workflow.yml:117]:
```suggestion
            make phpunit || (echo "A few of the unit tests were unsuccessful. Rerunning the unit test once again......" && make phpunit)
```

**Review by SagarGi — COMMENTED** [review]
If we are gonna retry the unit test only once when some flaky unit test fails then this method would be good instead of creating a separate step using other Github actions :+1:

**Review by individual-it — APPROVED** [review]

**Review by SagarGi — APPROVED** [review]

**Pull request merged**